airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -538,3 +393,51 @@ def stream_slices(
             # iterate over all parent records with current stream_slice
             for record in parent_records:
                 yield {"parent": record}
+
+
+@deprecated(version="3.0.0", reason="You should set backoff_strategies explicitly in HttpStream.get_backoff_strategy() instead.")
+class HttpStreamAdapterBackoffStrategy(BackoffStrategy):
+    def __init__(self, stream: HttpStream):
+        self.stream = stream
+
+    def backoff_time(
+        self, response_or_exception: Optional[Union[requests.Response, requests.RequestException]], **kwargs: Any
+    ) -> Optional[float]:
+        return self.stream.backoff_time(response_or_exception)  # type: ignore # noqa
+
+
+@deprecated(version="3.0.0", reason="You should set error_handler explicitly in HttpStream.get_error_handler() instead.")
+class HttpStreamAdapterHttpStatusErrorHandler(HttpStatusErrorHandler):
+    def __init__(self, stream: HttpStream, **kwargs):  # type: ignore # noqa
+        self.stream = stream
+        super().__init__(**kwargs)
+
+    def interpret_response(self, response_or_exception: Optional[Union[requests.Response, Exception]] = None) -> ErrorResolution:
+        if isinstance(response_or_exception, Exception):
+            return super().interpret_response(response_or_exception)
+        should_retry = self.stream.should_retry(response_or_exception)  # type: ignore # noqa
+        if should_retry:
+            return ErrorResolution(
+                response_action=ResponseAction.RETRY,
+                failure_type=FailureType.transient_error,
+                error_message="Response does not include an HTTP status code.",
+            )
+        else:
+            if response_or_exception.ok:  # type: ignore # noqa
+                return ErrorResolution(
+                    response_action=ResponseAction.SUCCESS,
+                    failure_type=None,
+                    error_message=None,
+                )
+            if self.stream.raise_on_http_errors:
+                return ErrorResolution(
+                    response_action=ResponseAction.FAIL,
+                    failure_type=FailureType.transient_error,
+                    error_message="Response does not include an HTTP status code.",
+                )
+            else:
+                return ErrorResolution(
+                    response_action=ResponseAction.IGNORE,
+                    failure_type=FailureType.transient_error,
+                    error_message="Response does not include an HTTP status code.",
```

Comment:
is this error_message right? `response_or_exception` is a `requests.Response` so I'd expect an HTTP status code

Reply:
updated with status code